lib/pkg.ml

```diff
@@ -465,6 +464,7 @@ let distrib_archive ~dry_run ~keep_dir ~include_submodules p =
   else Ok ())
   >>= fun () ->
   distrib_prepare ~dry_run ~dist_build_dir ~version >>= fun () ->
+  assert_tag_exists ~dry_run repo_vcs tag >>= fun () ->
```

Comment:
Why did you move this line around?

Reply:
The problem is that the test checks for `assert_tag_exists` before it runs `dune subst`.

Before the test it always succeeded because `assert_tag_exists` only tested whether whatever commit it got is a commit (basically a no-op) and then it ran `dune subst` which failed with the error message that the tests expect.

After fixing the problem arises that `assert_tag_exists` actually *fails* here, because there is no tag (given this is `check` it rightfully doesn't create any tag), so it exits with the error message that the tag is missing instead.

Hence I moved it around. I am not particularly happy about it because the function does everything in a sort of random order and the expect tests are very strongly coupled to which order it does things in.

Reply:
We should really avoid making changes like this just to get the tests to pass.

Which test was failing? From what I understood it's related to the check command and should work without a tag (which actually sounds good). I think we should look into this, eventually exposing an intermediate function that builds a distrib archive for any commit-ish (used by check) and one above it that runs all the proper checks (used by distrib).

Then we wouldn't have to change that code, and `check` would do the right thing. How does that sound?

Reply:
The test that fails is `tests/bin/distrib-name` where it uses `dune-release distrib` to check that `dune-release` will fail when `dune subst` can't find the name of a project. Without that change `dune subst` is never called because it fails earlier that the tag does not exist.

https://github.com/ocamllabs/dune-release/blob/f3f38cc463eaa90d44661f8fa41c8eccc55ea2c7/tests/bin/distrib-name/run.t#L56-L64

What the code on `main` does *is* the right behaviour — Calling `dune subst` will `git describe --always --dirty` which will write basically something somewhat random into the files, not the tag that we want.

Upon writing this I realized I was barking up the wrong tree. The code had the right behaviour and what I need to do is to fix the test to actually tag and then it all makes sense again.

Your suggestion to make an intermediate function is a good one though.